synapse/appservice/__init__.py

```diff
@@ -188,14 +158,14 @@ async def _matches_user(self, event, store):
         if not store:
             return False
 
-        does_match = await self._matches_user_in_member_list(event.room_id, store)
+        does_match = await self.matches_user_in_member_list(event.room_id, store)
         return does_match
 
     @cached(num_args=1, cache_context=True)
-    async def _matches_user_in_member_list(self, room_id, store, cache_context):
-        member_list = await store.get_users_in_room(
-            room_id, on_invalidate=cache_context.invalidate
-        )
+    async def matches_user_in_member_list(
+        self, room_id: str, store, cache_context: _CacheContext
+    ):
```

Comment:
Please add a type for `store` and a return type (`bool` it looks like).

It looks like the cache context is no longer being used, should the `cached` decorator be updated to stop providing it?

Since we're making this "public", please add a docstring.